src/makeCalculator.js

```diff
@@ -36,8 +36,35 @@
  *
  * @return {object}
  */
+
 function makeCalculator() {
-  // write code here
-}
+  const calculator = {
+    result: 0,
+    add(number) {
+      this.result += number;
+    },
+    subtract(number) {
+      this.result -= number;
+    },
+    multiply(number) {
+      this.result *= number;
+    },
+    divide(number) {
+      this.result /= number;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      callback.bind(calculator)(number);
```

Comment:
может просто 
```suggestion
      callback(number);
```
зачем усложнять?